src/lib/ThriftClient.js

```diff
@@ -0,0 +1,31 @@
+const thrift = require('thrift')
+const ExtensionManager = require('./thrift/ExtensionManager.js')
+const Types = require('./thrift/osquery_types.js')
+
+class ThriftClient {
+  static getInstance(opts) {
+    if (!this.instance) {
+      this.instance = new ThriftClient(opts)
+    }
+    return this.instance
+  }
+
+  constructor(opts = { path: '/Users/' + process.env.USER + '/.osquery/shell.em'}) {
+    const path = opts.path || '/Users/' + process.env.USER + '/.osquery/shell.em'
```

Comment:
Shouldn't have to do this since you've got the default in the arguments.

Also, same comment as above: should this be in appData?

Reply:
Good catch, this is a remnant from an earlier setup where I wasn't manually setting the socket path. Will remove this